app/actions/ActionsProvider.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package actions
+
+import config.{AppConfig, ErrorHandler}
+import models.{AuthorisationRequest, UserSessionDataRequest}
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+import services.CISSessionService
+import utils.InYearUtil
+import utils.UrlUtils.decoded
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ActionsProvider @Inject()(authAction: AuthorisedAction,
+                                cisSessionService: CISSessionService,
+                                errorHandler: ErrorHandler,
+                                inYearUtil: InYearUtil,
+                                appConfig: AppConfig
+                               )(implicit ec: ExecutionContext) {
+
+  def inYear(taxYear: Int): ActionBuilder[AuthorisationRequest, AnyContent] = authAction.andThen(inYearActionBuilder(taxYear))
```

Comment:
I have not tested those deliberately (I mentioned in the group review), as the inYear methods are not quite implemented yet. The method I use is tested in the controller, and as mentioned I will move that with the next story.

Reply:
Yes, but It might be a mock controller. I have not thought about it yet. It is the same for all the controllers so that is why it needs to be taken out separately.

Reply:
okay